Brokerages/GDAX/GDAXBrokerage.cs

```diff
@@ -0,0 +1,349 @@
+                {
+                    var orders = JsonConvert.DeserializeObject<Messages.Order[]>(response.Content);
+                    foreach (var item in orders)
+                    {
+                        Order order = null;
+                        if (item.Type == "market")
+                        {
+                            order = new MarketOrder { Price = item.Price };
+                        }
+                        else if (item.Type == "limit")
+                        {
+                            order = new LimitOrder { LimitPrice = item.Price };
+                        }
+                        else if (item.Type == "stop")
+                        {
+                            order = new StopMarketOrder { StopPrice = item.Price };
+                        }
+                        else
+                        {
+                            Log.Error("GDAXBrokerage.GetOpenOrders: Unsupported order type returned from brokerage" + item.Type);
+                            continue;
+                        }
+
+                        order.Quantity = item.Side == "sell" ? -item.Size : item.Size;
+                        order.BrokerId = new List<string> { item.Id.ToString() };
+                        order.Symbol = ConvertProductId(item.ProductId);
+                        order.Time = DateTime.UtcNow;
+                        order.Status = ConvertOrderStatus(item);
+                        order.Price = item.Price;
+                        list.Add(order);
+                    }
+                }
+            }
+            catch (Exception)
+            {
+                throw;
+            }
+
+            foreach (Order item in list)
+            {
+                if (item.Status.IsOpen())
+                {
+                    var cached = this.CachedOrderIDs.Where(c => c.Value.BrokerId.Contains(item.BrokerId.First()));
+                    if (cached.Any())
+                    {
+                        this.CachedOrderIDs[cached.First().Key] = item;
+                    }
+                }
+            }
+
+            return list;
+
+        }
+
+        /// <summary>
+        /// Gets all open positions
+        /// </summary>
+        /// <returns></returns>
+        public override List<Holding> GetAccountHoldings()
+        {
+            var list = new List<Holding>();
+
+            var req = new RestRequest("/orders?status=active", Method.GET);
+            GetAuthenticationToken(req);
+            var response = RestClient.Execute(req);
+            if (response != null)
+            {
+                foreach (var item in JsonConvert.DeserializeObject<Messages.Order[]>(response.Content))
+                {
+
+                    decimal conversionRate;
+                    if (!item.ProductId.EndsWith("USD", StringComparison.InvariantCultureIgnoreCase))
+                    {
+
+                        var baseSymbol = (item.ProductId.Substring(0, 3) + "USD").ToLower();
```

Comment:
We'll soon need to figure out how to solve the crypto-symbol issue =) Lets kick that one down the road a little....